api/v1alpha1/dnsrecord_types.go

```diff
@@ -33,10 +33,19 @@ const HttpsProtocol HealthProtocol = "HTTPS"
 // By default this health check will be applied to each unique DNS A Record for
 // the listeners assigned to the target gateway
 type HealthCheckSpec struct {
-	Endpoint         string          `json:"endpoint,omitempty"`
-	Port             *int            `json:"port,omitempty"`
-	Protocol         *HealthProtocol `json:"protocol,omitempty"`
-	FailureThreshold *int            `json:"failureThreshold,omitempty"`
+	// Should start with "?" or "|", contain only allowed for URLs characters and end with alphanumeric char or "/"
```

Comment:
// Must start with "?" or "/", contain only valid URL characters and end with alphanumeric char or "/"